cpp/src/arrow/pretty_print.cc

```diff
@@ -19,7 +19,7 @@
 #include <cstdint>
 #include <iostream>
 #include <memory>
-#include <sstream>
+#include <sstream>  // IWYU pragma: keep
```

Comment:
I keep seeing these in the code base. What are they for? (the "IWYU pragma" comments)

Reply:
Have you run include-what-you-use before? The problem is that the C++ standard library is a little bit different on Windows, and so IWYU will say to remove a header sometimes and that will break the Windows build. So the pragmas are about having clean IWYU runs

Reply:
I see, thanks. I think I ran it before, but I don't remember any specific issues.

Reply:
I think it's better to deal with this particular issue using mappings files (https://github.com/apache/arrow/tree/master/cpp/build-support/iwyu/mappings), so at some point we should try to clean up the "keep" pragmas using those